File: VB/DevExpress.Xpf.Grid.Extensions/DevExpress.Xpf.Grid.Extensions/Exceptions.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpf.Grid
Namespace BandedViewExtension
	Public Class InvalidColumnPropertyValueException
		Inherits InvalidOperationException
		Private privateGridColumn As ColumnBase
		Public Property GridColumn() As ColumnBase
			Get
				Return privateGridColumn
			End Get
			Private Set(ByVal value As ColumnBase)
				privateGridColumn = value
			End Set
		End Property
		Private privateGridColumnPropertyName As String
		Public Property GridColumnPropertyName() As String
			Get
				Return privateGridColumnPropertyName
			End Get
			Private Set(ByVal value As String)
				privateGridColumnPropertyName = value
			End Set
		End Property
		Public Overrides ReadOnly Property Message() As String
			Get
				Return "You cannot set the " & GridColumnPropertyName & "property for the GridColumn when BandedBehavior is used."
			End Get
		End Property
		Public Sub New(ByVal column As ColumnBase, ByVal propertyName As String)
			GridColumn = column
			GridColumnPropertyName = propertyName
		End Sub
	End Class
End Namespace

File: CS/DevExpress.Xpf.Grid.Extensions/DevExpress.Xpf.Grid.Extensions/Exceptions.cs
using System;
using DevExpress.Xpf.Grid;
namespace BandedViewExtension {
    public class InvalidColumnPropertyValueException : InvalidOperationException {
        public ColumnBase GridColumn { get; private set; }
        public string GridColumnPropertyName { get; private set; }
        public override string Message {
            get { return "You cannot set the " + GridColumnPropertyName + "property for the GridColumn when BandedBehavior is used."; }
        }
        public InvalidColumnPropertyValueException(ColumnBase column, string propertyName) {
            GridColumn = column;
            GridColumnPropertyName = propertyName;
        }
    }
}